reference/employer/openapi.yaml

```diff
@@ -0,0 +1,1083 @@
+        '200':
+          description: Pay statement data
+          content:
+            application/json:
+              examples:
+                Success:
+                  value:
+                    responses:
+                      - payment_id: fc8b024e-d373-4c9c-80fc-f1625383d142
+                        code: 200
+                        body:
+                          paging:
+                            count: 25
+                            offset: 10
+                          pay_statements:
+                            - individual_id: be7b048c-a6f3-4194-a017-2f537d4f3565
+                              type: regular
+                              payment_method: check
+                              total_hours: 80
+                              gross_pay:
+                                amount: 230000
+                                currency: usd
+                              net_pay:
+                                amount: 180000
+                                currency: usd
+                              earnings:
+                                - type: salary
+                                  name: Salary
+                                  amount: 230000
+                                  currency: usd
+                              taxes:
+                                - type: state
+                                  name: State Withholding - OR
+                                  employer: false
+                                  amount: 0
+                                  currency: usd
+                              employee_deductions:
+                                - name: Pre-Tax 401k
+                                  amount: 50000
+                                  currency: usd
+                                  pre_tax: true
+                              employer_contributions:
+                                - name: Employee Medical Insurance
+                                  amount: 23272
+                                  currency: usd
+              schema:
+                type: object
+                properties:
+                  responses:
+                    type: array
+                    items:
+                      type: object
+                      properties:
+                        payment_id:
+                          type: string
+                        code:
+                          type: integer
+                        body:
+                          type: object
+                          properties:
+                            paging:
+                              $ref: '#/components/schemas/Paging'
+                            pay_statements:
+                              type: array
+                              description: The array of pay statements for the current payment.
+                              items:
+                                type: object
+                                properties:
+                                  individual_id:
```

Comment:
Everything but `individual_id` is nullable

Reply:
```
const responseSchema = Joi.object({
  paging: Joi.object({
    count: Joi.number(),
    offset: Joi.number(),
  }),
  pay_statements: Joi.array().items(
    Joi.object({
      individual_id: Joi.string().guid(),
      type: Joi.string().valid(
        'regular_payroll',
        'off_cycle_payroll',
        'one_time_payment'
      ),
      payment_method: Joi.string().valid('direct_deposit', 'check').allow(null),
      gross_pay: commonSchemas.amount,
      net_pay: commonSchemas.amount,
      total_hours: Joi.number().allow(null),
      earnings: Joi.array().items(
        Joi.alternatives(
          // we want either name/type otherwise the data is useless for the developer
          commonSchemas.amount.keys({
            type: Joi.string()
              .valid(
                'salary',
                'wage',
                'reimbursement',
                'severance',
                'pto',
                'sick',
                'bonus',
                'commission',
                'tips',
                '1099',
                'overtime',
                'double_overtime',
                'other'
              )
              .allow(null),
            name: Joi.string(),
          }),
          commonSchemas.amount.keys({
            type: Joi.string(),
            name: Joi.string().allow(null),
          })
        )
      ),
      taxes: Joi.array().items(
        commonSchemas.amount.keys({
          employer: Joi.boolean(),
          name: Joi.string(),
          type: Joi.string()
            .valid('state', 'federal', 'local', 'medicare', 'fica')
            .allow(null),
        })
      ),
      employee_deductions: Joi.array()
        .items(
          commonSchemas.amount.keys({
            name: Joi.string(),
            pre_tax: Joi.boolean(),
          })
        )
        .allow(null),
      employer_contributions: Joi.array()
        .items(
          commonSchemas.amount.keys({
            name: Joi.string(),
          })
        )
        .allow(null),
    })
  ),
});
```